src/views/List.jsx

```diff
@@ -1,13 +1,32 @@
+import { useState, useEffect } from 'react';
 import { ListItem } from '../components';
 
 export function List({ data }) {
+	const [searchItem, setSearchItem] = useState('');
+	const [filteredItems, setFilteredItems] = useState(data);
+
+	useEffect(() => {
+		setFilteredItems(
+			data.filter((item) =>
+				item.name.toLowerCase().includes(searchItem.toLowerCase()),
+			),
+		);
+	}, [searchItem, data]);
+
```

Comment:
I don't see a need for the state here and also filtering isn't a side effect therefore I also don't see a need for the useEffect. 

Simply filtering directly works well, of course, if a user hasn't entered any letters the whole data will be returned cause there is nothing to filter against, then when they start adding items it filters.😊

```
const filteredItems = data?.filter((item) => item?.name.toLowerCase().includes(searchItem.toLowerCase()))
```

This worked just the same for me.

Notice how I used the optional changing because even tho we fetch data as soon as the user opens the app, something might happen and the data isn't available at the point when the user is trying to filter, so that helps to make sure the app doesn't crash when data isn't available.